src/providers/network-connection-provider.test.tsx

```diff
@@ -0,0 +1,216 @@
+import React from "react";
+import { act, cleanup, render } from "@testing-library/react";
+import { NetworkConnectionProvider } from "./network-connection-provider";
+import {
+    NetworkConnection,
+    NetworkInformationUtils,
+} from "andculturecode-javascript-core";
+import { useNetworkConnection } from "../hooks/use-network-connection";
+import { FactoryType } from "../tests/factories/factory-type";
+import { Factory } from "rosie";
+
+// -----------------------------------------------------------------------------------------
+// #region Types
+// -----------------------------------------------------------------------------------------
+
+type TypeFromKey<Type, Key extends keyof Type> = Type[Key];
```

Comment:
Suggestion: The more I think about this it might be better named `TypeOfKey` instead of `From`

Reply:
I like tha.